sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/InjectRuntimeFilter.scala

```diff
@@ -209,15 +210,35 @@ object InjectRuntimeFilter extends Rule[LogicalPlan] with PredicateHelper with J
       REGEXP_EXTRACT_FAMILY, REGEXP_REPLACE)
   }
 
-  private def isProbablyShuffleJoin(left: LogicalPlan,
-      right: LogicalPlan, hint: JoinHint): Boolean = {
-    !hintToBroadcastLeft(hint) && !hintToBroadcastRight(hint) &&
-      !canBroadcastBySize(left, conf) && !canBroadcastBySize(right, conf)
+  // Whether it is a shuffle join or not should be based on the actual left and
+  // right table. For some join like left outer join, it will be a shuffle join
+  // even if left side table size is smaller than broadcast threshold.
```

Comment:
Could we simplify this comment?
`Whether it is a shuffle join or not based on the join type, the left size and right size.`